src/KeyUnit.cpp

```diff
@@ -60,14 +60,13 @@ void KeyUnit::uninstall(const QString& packageName)
 
 bool KeyUnit::processDataStream(int key, int modifier)
 {
-    bool isAtLeastOneMatch = false;
     for (auto keyObject : mKeyRootNodeList) {
         if (keyObject->match(key, modifier)) {
-            isAtLeastOneMatch = true;
+            return true;
         }
     }
 
-    return isAtLeastOneMatch;
+    return false;
```

Comment:
Hang on, the Wiki currently says:
> Note: Mudlet by design allows duplicate names - so calling permKey [or tempkey] with the same name [code and modifier] will keep creating new keys. You can check if a key already exists with the exists() function.

It does not say "but only the first one of any _code and modifier combination_ of either permanent or temporary Keys" will work and the remainder will be ignored...!  So if you create a key of either type and drop in buggy lua code and then correct it and call the same command again you will have two instances and only the first, buggy one is going to be (and fail to) run.  That is not conforming to "principle of least surprise" IMHO.